lib/tasks/data_key_validation.rake

```diff
@@ -0,0 +1,52 @@
+
+namespace :data_key_validation do
+  desc "Validates the value of ENV['CONJUR_DATA_KEY']"
+  # This task validate the master data key which is stored in `ENV['CONJUR_DATA_KEY']`
+  # Once the db is created, a secret will be created using the master key.
+  # Each time the server is restarted this task will attempt to retrieve the
+  # secret, if the provided key does not match the original master the secret retrieval
+  # will fail.
+  #
+  # What's the 'environment' task in Rake?
+  # the :validate => :environment means "do environment before validate".
+  # You can get access to your models, and in fact, your whole environment
+  # by making tasks dependent on the environment task.
+  task :validate => [:environment] do
+    data_key = ENV['CONJUR_DATA_KEY']
+
+    if data_key.nil? || data_key.empty?
+      raise Errors::System::EmptyConjurDataKey
+      exit 1
+    end
+
+    # try get the master key validator resource
+    resource_id = "!:!:data-key-validator-resource"
+
+    unless Resource[resource_id]
+      # create the master key validator resource
+      # and store a secret using the master key
+      role_id = "!:!:data-key-validator"
+
+      begin
+        Role.create(role_id: role_id)
+        Resource.create(resource_id: resource_id, owner_id: role_id)
+        Secret.create(resource_id: resource_id, value: "Don't Panic.")
+      rescue => e
+        $stderr.puts "Error creating a resource for rake: data_key_validation, Error: #{e.inspect}"
+      end
+    else
+      begin
+        # Attempt to retrieve the secret encrypt with the original data key.
+        # A failure to decrypt the value will indicate the key is invalid.
+        Resource[resource_id].secret.value
+      rescue => e
+        if e.inspect == 'OpenSSL::Cipher::CipherError'
```

Comment:
Is there a possibility that this error type could be raised not because of the data key being invalid? I want to make sure there will not be a case where we write a misleading error message.

Reply:
Only if the key is Invalid.